File: gen_scripts/perl/f30.loop.pl
#!/usr/bin/env perl -w
# spin loop construction with last to exiting
while (1) {
  print "Enter your name (quit to exit): "; # prompt user
  chomp($answer = <>);                      # git input and trim newline
  last if $answer eq "quit";                # exit loop if user wants to quit
  print "Hello $answer!\n";                  # output results as not exiting
}

File: lessons/win_scripts/wsh.jscript/scripts/f30.loop.js
// spin loop as condition is always true
//  use break to exit out of loop
do {
  // prompt user
  WScript.stdout.write("Enter your name (quit to exit): ");
  answer = WScript.stdin.readline();         // get input

  // exit loop is user quits
  if (answer == "quit") break

  // output result as not exiting
  WScript.Echo("Hello " + answer + "!");
} while (true)
